tests/ui/test_ui_tools.py

```diff
@@ -1654,6 +1669,7 @@ def test_soup2markup(self, content, markup):
         self.model.server_url = SERVER_URL
         # NOTE Absence of previous (last) message should not affect markup
         msg_box = MessageBox(message, self.model, None)
+        msg_box.link_index = 1
```

Comment:
It was necessary as I was initializing the variable in `soup2markup`'s caller method. That said, I gather it better belongs to the `__init__`.

Re `len(message_links)`, I am not sure if this would work here.

Reply:
By my reading, `link_index` is really 'next link index', so equivalent to `len(message_links)+1`? So we could avoid needing to synchronize two variables.

Reply:
Thanks for the tip! I have implemented it in the latest update.